tests/model_forms/tests.py

```diff
@@ -2012,6 +2013,38 @@ def test_runtime_choicefield_populated(self):
             ),
         )
 
+    @isolate_apps("model_forms")
+    def test_callable_choices_are_lazy(self):
+        call_count = 0
+
+        def get_animal_choices():
+            nonlocal call_count
+            call_count += 1
+            return [("LION", "Lion"), ("ZEBRA", "Zebra")]
+
+        class ZooKeeper(models.Model):
+            animal = models.CharField(
+                blank=True,
+                choices=get_animal_choices,
+                max_length=5,
+            )
+
+        class ZooKeeperForm(forms.ModelForm):
+            class Meta:
+                model = ZooKeeper
+                fields = ["animal"]
+
+        self.assertEqual(call_count, 0)
```

Comment:
I was hoping we could use a Mock.mock(wraps=...) here but it seems it does not work with dunder methods so the call is not recorded. Is this why you chose to use a plain integer count?

Reply:
Honestly I did this because I was checking at what point things got called after definition, etc and it was a simple next step from the provided reproducer that used a boolean. If it can be made to use the approach you mentioned, that's fine, but this is also simple and self-contained.

Reply:
I tried to change to use mock but it fails as if the mock was never called (but it is!)
After some investigation, this may be related with https://github.com/python/cpython/issues/69783

So I think that leaving it as proposed is OK. Thanks!